### PR TITLE
migrate to BL2_EL3 on both hikey and hikey960

### PR DESCRIPTION
**Pull request by hzhuang1**

### CODE REVIEW

File: plat/hisilicon/hikey/aarch64/hikey_common.c [hunk 1]
```diff
@@ -28,7 +28,7 @@
 					TSP_SEC_MEM_SIZE,		\
 					MT_MEMORY | MT_RW | MT_SECURE)
 
-#if LOAD_IMAGE_V2
+#if LOAD_IMAGE_V2 && (HIKEY_TSP_RAM_LOCATION_ID == HIKEY_SRAM_ID)
```
Comment (vchong) [inline plat/hisilicon/hikey/aarch64/hikey_common.c]: `HIKEY_OPTEE_PAGEABLE_LOAD_*` are not defined for SRAM use atm, so maybe just remove `MAP_OPTEE_PAGEABLE` from `hikey_mmap` until they're defined for SRAM use?

Same for below.

File: plat/hisilicon/hikey960/aarch64/hikey960_common.c [hunk 2]
```diff
@@ -41,7 +41,7 @@
 					TSP_SEC_MEM_SIZE,		\
 					MT_MEMORY | MT_RW | MT_SECURE)
 
-#if LOAD_IMAGE_V2
+#if LOAD_IMAGE_V2 && (HIKEY960_TSP_RAM_LOCATION_ID == HIKEY960_SRAM_ID)
```
Comment (vchong) [inline plat/hisilicon/hikey960/aarch64/hikey960_common.c]: `HIKEY960_OPTEE_PAGEABLE_LOAD_*` are not defined for SRAM use atm, so maybe just remove `MAP_OPTEE_PAGEABLE` from `hikey960_mmap` until they're defined for SRAM use?

Same for below.

File: plat/hisilicon/hikey960/hikey960_bl2_setup.c [hunk 3]
```diff
@@ -47,98 +45,13 @@
...
-}
-
-/*******************************************************************************
- * Populate the extents of memory available for loading SCP_BL2 (if used),
- * i.e. anywhere in trusted RAM as long as it doesn't overwrite BL2.
- ******************************************************************************/
-void bl2_plat_get_scp_bl2_meminfo(meminfo_t *scp_bl2_meminfo)
-{
-	hikey960_init_ufs();
-	hikey960_io_setup();
```
Comment (vchong) [inline plat/hisilicon/hikey960/hikey960_bl2_setup.c]: When I added support for `LOAD_IMAGE_V2`, I moved `hikey960_init_ufs()` and `hikey960_io_setup()` to `plat_get_bl_image_load_info()` in `hikey960_image_load.c`. Iiuc, these 2 functions will then be called multiple times unnecessarily by `bl2_load_images()` in `bl2_image_load_v2.c`. In retrospect, perhaps they should go to `bl2_early_platform_setup()` or `bl2_platform_setup()` in `hikey960_bl2_setup.c` instead?

Comment (hzhuang1) [reply plat/hisilicon/hikey960/hikey960_bl2_setup.c]: Yes, it should move to bl2_platform_setup(). I didn't notice that you moved it into plat_get_bl_image_load_info(). I'll fix it.

File: plat/hisilicon/hikey/aarch64/hikey_common.c (hunk 1 above)
Comment (hzhuang1) [reply plat/hisilicon/hikey/aarch64/hikey_common.c]: OK. I'll remove it.

File: plat/hisilicon/hikey960/aarch64/hikey960_common.c (hunk 2 above)
Comment (hzhuang1) [reply plat/hisilicon/hikey960/aarch64/hikey960_common.c]: OK. I'll remove it.

**Comment by hzhuang1** [comment]
I hope this pull request could be held for a while. It's only for review, not merge right now. 
Our guys haven't prepared to migrate yet.

**Comment by davidcunado-arm** [comment]
@hzhuang1 
Yep no problem - do you have a rough timeline? If it's going to be more than a couple of weeks, we could close and then re-open when you're ready?

**Comment by hzhuang1** [comment]
@davidcunado-arm We need at least one week. So let me close it first.

File: docs/plat/hikey960.rst [hunk 4]
```diff
@@ -146,7 +146,7 @@ Boot UEFI in recovery mode
        # The content of config file
        ./sec_usb_xloader.img 0x00020000
        ./sec_uce_boot.img 0x6A908000
-       ./l-loader.bin 0x1AC00000
+       ./recovery.bin 0x1AC00000
```
Comment (vchong) [inline docs/plat/hikey960.rst]: In L157 below, please change to `Run the command to download recovery.bin into HiKey960`. Thanks!

File: docs/plat/hikey960.rst [hunk 5]
```diff
@@ -130,13 +131,12 @@ Boot UEFI in recovery mode
...
    `link <https://github.com/96boards-hikey/tools-images-hikey960>`__
 
--  Generate l-loader.bin.
+-  Prepare recovery binary.
 
    .. code:: shell
 
        $cd tools-images-hikey960
-       $ln -sf ${BUILD_PATH}/l-loader/l-loader.bin
-       $ln -sf ${BUILD_PATH}/l-loader/fip.bin
```
Comment (vchong) [inline docs/plat/hikey960.rst]: Keep this line?

File: docs/plat/hikey960.rst (hunk 4 above)
Comment (hzhuang1) [reply docs/plat/hikey960.rst]: ok

File: docs/plat/hikey960.rst (hunk 5 above)
Comment (hzhuang1) [reply docs/plat/hikey960.rst]: OK

Comment (vchong) [reply docs/plat/hikey960.rst]: I mean the symlink to fip.bin.

**Comment by hzhuang1** [comment]
Replace it in https://github.com/ARM-software/arm-trusted-firmware/pull/1277. Since it's refreshed, I have to create it a new pull request.

**Pull request closed without merging**